python/pyarrow/_dataset.pyx

```diff
@@ -1053,6 +1053,39 @@ cdef class FileFragment(Fragment):
             self.partition_expression
         )
 
+    def open(self):
+        """
+        Open a NativeFile of the buffer or file viewed by this fragment.
+        """
+        cdef:
+            shared_ptr[CFileSystem] c_filesystem
+            shared_ptr[CRandomAccessFile] opened
+            c_string c_path
+            NativeFile out = NativeFile()
+
+        if self.buffer is not None:
+            return pa.io.BufferReader(self.buffer)
```

Comment:
Is this exercised by the tests somewhere?

Reply:
Good catch. :)